fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/keyspace/ExtendedDirectoryLayer.java

```diff
@@ -67,31 +66,42 @@
     private final Subspace contentSubspace;
 
     /**
-     * Create an extended directory layer. This constructor utilizes blocking calls and should not
-     * be used in an asynchronous context.
+     * Create an extended directory layer.
      *
      * @param context the context that will be used to resolve the provided path into the subspace at which
      *   the directory layer will be created. This context is only used during the construction of the
      *   this class is not subsequently used
      * @param path the path at which the directory layer should store its mappings.
+     * @deprecated use {@link #ExtendedDirectoryLayer(FDBDatabase, ResolvedKeySpacePath)} instead
      */
+    @Deprecated
+    @API(API.Status.DEPRECATED)
     public ExtendedDirectoryLayer(@Nonnull FDBRecordContext context, @Nonnull KeySpacePath path) {
-        this(context.getDatabase(), context, path);
+        this(context.getDatabase(), path, path.toResolvedPathAsync(context));
+    }
+
+    /**
+     * Create an extended directory layer.
+     *
+     * @param database database that will be used when resolving values
+     * @param path the path at which the directory layer should store its mappings.
+     */
+    public ExtendedDirectoryLayer(@Nonnull FDBDatabase database, @Nonnull ResolvedKeySpacePath path) {
+        this(database, path.toPath(), CompletableFuture.completedFuture(path));
     }
 
     private ExtendedDirectoryLayer(@Nonnull FDBDatabase database,
-                                   @Nullable FDBRecordContext context,
-                                   @Nullable KeySpacePath path) {
-        super(database, path);
-        if (path == null) {
+                                   @Nullable KeySpacePath path,
```

Comment:
Yes, it is required by the:

    super(database, path, resolvedPathFuture);

below. It will only be needed until the deprecated constructor, above, is removed.